app/upgrades/v17/upgrades.go

```diff
@@ -303,3 +309,61 @@ func AddRateLimitToOsmosis(ctx sdk.Context, k ratelimitkeeper.Keeper) error {
 
 	return nil
 }
+
+// Migrate the unbonding records
+// UserUnbondingRecords previously only used Native Token Amounts, we now want to use StTokenAmounts
+// Similarly, we should modify HostZoneUnbondingRecords to NOT use NativeTokenAmounts prior to unbonding being initiated
+func MigrateUnbondingRecords(ctx sdk.Context, k stakeibckeeper.Keeper) error {
+	for _, epochUnbondingRecord := range k.RecordsKeeper.GetAllEpochUnbondingRecord(ctx) {
+		for _, hostZoneUnbonding := range epochUnbondingRecord.GetHostZoneUnbondings() {
+			// if the tokens have unbonded, we don't want to modify the record
+			// this is because we won't be able to estimate a redemption rate
+			if hostZoneUnbonding.Status == recordtypes.HostZoneUnbonding_CLAIMABLE {
+				continue
+			}
+			// similarly, if there aren't any tokens to unbond, we don't want to modify the record
+			// as we won't be able to estimate a redemption rate
+			if hostZoneUnbonding.NativeTokenAmount == sdkmath.ZeroInt() {
+				continue
+			}
+
+			// Calculate the estimated redemption rate
+			nativeTokenAmountDec := sdk.NewDecFromInt(hostZoneUnbonding.NativeTokenAmount)
+			stTokenAmountDec := sdk.NewDecFromInt(hostZoneUnbonding.StTokenAmount)
+			// this estimated rate is the amount of stTokens that would be received for 1 native token
+			// e.g. if the rate is 0.5, then 1 native token would be worth 0.5 stTokens
+			estimatedStTokenConversionRate := stTokenAmountDec.Quo(nativeTokenAmountDec)
+
+			// store if the unbonding has not been initiated
+			unbondingNotInitiated := hostZoneUnbonding.Status == recordtypes.HostZoneUnbonding_UNBONDING_QUEUE
+
+			// Loop through User Redemption Records and insert an estimated stTokenAmount
```

Comment:
Do we want update records that have the following statuses all in the same way?

1. `HostZoneUnbonding_UNBONDING_QUEUE`
2. `HostZoneUnbonding_UNBONDING_IN_PROGRESS` <= is there a risk we are going to catch any of these in flight?
3. `HostZoneUnbonding_EXIT_TRANSFER_QUEUE`
4. `HostZoneUnbonding_EXIT_TRANSFER_IN_PROGRESS`

Reply:
1. Yep, makes sense. 
2. Thinking about this a bit more, not only will nothing bad happen here, in fact we _need_ to migrate these or their undelegate callbacks will fail because in `UndelegateCallback` we invoke `SetNativeTokensToZeroInUnbondingRecords` which references their `stTokenAmount` attribute. If they have no `stTokenAmount` that callback will return err. 
3. Agreed, we should migrate it. And feels safe: I can't find a reference to stTokenAmount in any code post-UndelegateCallback. Imo this is the same as 4 but different than 2 (2 _needs_ the migration to avoid erroring).
4. Yeah I think we should migrate it. Imo this is the same as 3 but different than 2 (2 _needs_ the migration to avoid erroring).

Reply:
Ah yeah great call re:(2)! Let's keep this as is then, seems like the migration is good for all of the cases

Reply:
Agreed let's keep it as is

Reply:
Leaving this open so @sampocs and @asalzmann can think through each case with this context as well. Feel free to resolve after.